### ISSUE TITLE
Glyphicons via node_modules

### ISSUE THREAD

**Issue by pmonty** [fuse-box/fuse-box#1222]
So have included bootstrap sass from node_modules in my stylesheet:

```
$icon-font-path: '~bootstrap-sass/assets/fonts/bootstrap/';
@import '~bootstrap-sass/assets/stylesheets/_bootstrap.scss';
@import '~animate.sass/scss/animate.scss';
```

The styles are definitely getting bundles but the font files are not.

[They are included in _glyphicons.scss](https://github.com/twbs/bootstrap-sass/blob/51486a8bd836d32b9f413e911ed83b433ef4ad39/assets/stylesheets/bootstrap/_glyphicons.scss#L14) file in bootstrap-sass and they don't appear in the css-resources folder.

I also have the following lines in my main scss file
```
// Typography
@font-face {
    font-family: 'MSR100';
    src: url('https://s3-us-west-2.amazonaws.com/s.cdpn.io/181511/museosans-rounded-100-webfont.woff2') format('woff2'), url('https://s3-us-west-2.amazonaws.com/s.cdpn.io/181511/museosans-rounded-100-webfont.woff') format('woff');
    font-weight: normal;
    font-style: normal;
}
```

I believe those ones are working but the ones inside bootstrap-sass don't.

My relevant Fuse configuration settings:

```
[
    SassPlugin({
        importer: true
    }),
    CSSResourcePlugin({ 
         dist: 'dist/assets/css-resources',
         macros: {
               icons: `${__dirname}/src/~/wholesalers/${this.wholesaler}/images/icons/`,
               base: `${__dirname}/src/~/wholesalers/base/images`
         },
         resolve: (f) => `/assets/css-resources/${f}`
      }),
      CSSModules({ scopedName: '[local]'}),
      CSSPlugin(this.cssOptions)
],
```

Thought it might have been my CopyPlugin so added the woff, ttf, eot and svg file types to that but didn't work.

**Comment by jpike88** [comment]
I use this and it works good. [CSSResourcePlugin({ inline: true }), CSSPlugin()]

You seem to be doing a fair amount of directory funging/renaming here, why not simplify things a bit?

**Comment by nchanged** [comment]
@pmonty `~` is a very weird directory name. TIlde is a conventional `home` directory symbol

**Comment by pmonty** [comment]
I tried inline didn't work. I'll try rename the tilde folder when I get home but didn't work in m playground app that doesn't have it and has a simpler setup. Will let you guys know how it goes

**Comment by jpike88** [comment]
I don't know much about the particularities of your project, but from my understanding, if fusebox (or any bundler) isn't being utilised to bundle your entire application in one go, you're going to run into some very strange issues...

This is my fuse script, hopefullly this gives you some insight.

https://gist.github.com/jpike88/181ebabc369ad8678a466a9672fa8a41

**Comment by pmonty** [comment]
Tried without the tilde folder and still didn't work.

The css that is imported is converted to something similar like
```
src: url("css-resource/8934789747.eot")
```

but there is no eot file in css-resources folder

**Comment by pmonty** [comment]
Not sure if this is part of the issue...
```
Without `from` option PostCSS could generate wrong source map and will not find Browserslist config. Set it to CSS file path or to `undefined` to prevent this warning.
```

**Comment by pmonty** [comment]
Tested inline again and those .woff and .ttf etc. all convert to undefined in my css. So looks like there might be an issue with one of the plugins not sure if a bug or config issue?

Files all are in the right paths.

Here is [my fuse.ts](https://gist.github.com/pmonty/4343b376ef1edadcafad5847fb4a34f7) so far.

**Comment by jpike88** [comment]
Can you fire up a seperate repo with this issue? I'll need more than your fuse.ts to reproduce it

**Comment by pmonty** [comment]
[This suggestion](https://github.com/fuse-box/fuse-box/issues/944) seems to work. Though would be nice if it can be resolved inside sass/css. 

Doing so has caused a few weird css issues with some font sizes and so on so will take a closer look at those before closing this.

**Comment by jpike88** [comment]
You're using a more advanced scenario than me, I'm afraid I can't be much use at this point... it's a relatively young project, you may be hitting some rough edges that could use some smoothing out.

**Comment by pmonty** [comment]
So not all my packages are in my vendor bundle. 
```
return this.fuse.bundle('vendor').instructions(`~ app.tsx`);
```
jquery looks like it is bundled in there doesn't work. I added some shim configuration and it bundled it at the top of the file instead and that didn't work.

I also noticed bootstrap javascript is not bundled despite it being in the bootstrap-sass module. This would probably be what is causing the javascript/jquery not working issue since it is all bootstrap stuff.

Still got the glyphicon issue as well. If i import in typescript then I get another sourcemap. Shouldn't be necessary though as everything is imported in my stylesheets all mapped from the main file that I import at compile time. So not sure why I would need to import it a second time.

**Comment by pmonty** [comment]
Done some more research and tests. So bootstrap-sass does not have a /dist folder or index.js file. Sounds like Fuse-Box requires those if it is going to bundle them into the vendor.js bundle.

Not sure if there is a way to bundle this then but really should be a plugin kind of like webpacks ProvidePlugin or something that allows you to just specify things and bundle all the main javascript stuff accordingly.

**Comment by nchanged** [comment]
@pmonty I think it's better to import it via `typescript` and a CSS dist because of image resolution.

**Comment by pmonty** [comment]
Scratched some text in post above, so looks like this works and gets the icons in but the bootstrap jquery is still missing in the bundle.

```
import 'bootstrap-sass/assets/stylesheets/_bootstrap.scss';
import 'animate.sass/scss/animate.scss';
```

Some fo the css styling is off a little bit but will get our stylesheet expert to look at those smaller issues. 

Unfortunately bootstrap-sass doesn't have a dist :(

**Comment by nchanged** [comment]
@pmonty import it in typescript

```ts
import 'bootstrap/dist/css/bootstrap.css';
```

**Comment by pmonty** [comment]
Yeh that was one of the combinations I tried and it didn't work. Site looks the same with or without that line. So it isn't liking it for some reason.

**Comment by nchanged** [comment]
Do you have ` [CSSResourcePlugin({ inline: true }), CSSPlugin()]` that one on your plugin chain?

**Comment by pmonty** [comment]
```
                    [
                        SassPlugin({ 
                            importer: true 
                        }),
                        CSSResourcePlugin({ 
                            inline: true
                        }),
                        CSSPlugin(),
                    ],```

**Comment by nchanged** [comment]
@pmonty 
You handle only `.css`, I don't see a CSS handler
Add this additionally:
```
[CSSResourcePlugin({ inline: true }), CSSPlugin()]
```

**Comment by pmonty** [comment]
Don't need it but have decided to try and import everything through typescript and have done this...
```
import 'jquery/dist/jquery.min.js';
import '~/wholesalers/EA/styles/site.scss';
import 'bootstrap-sass/assets/stylesheets/_bootstrap.scss';
import 'bootstrap-sass/assets/javascripts/bootstrap.min.js';
```
but am getting 
`Uncaught Error: Bootstrap's JavaScript requires jQuery`
which is strange because jquery is before bootstrap...

**Comment by nchanged** [comment]
You need to shim jquery that's for sure

**Comment by pmonty** [comment]
Shimming fixed it and doing this works so far and all glyphicons work.
```
import 'bootstrap-sass/assets/stylesheets/_bootstrap.scss';
import 'bootstrap-sass/assets/javascripts/bootstrap.min.js';
```

There are a few broken things but we can probably deal with those few things, like turn some broken table into a react-table and some other overridden bootstrap stuff that is slightly off. Stuff i expected would break when entirely changing bundler on a rather complex and large project.

There is one issue left involving react-router I believe. I'll close this issue now.

TLDR;
needed to shim jquery in my fuse.ts and include bootstrap-sass stylesheet and bootstrap.min.js in my main app.tsx file. The animate.sass npm module that I imported via sass is fine there but the bootstrap-sass had to be imported via typescript because of the inline resources/glyhicons
